app_utils.py

```diff
@@ -0,0 +1,33 @@
+import functools
+from flask_jwt_extended import jwt_required, get_jwt_identity
+from flask import abort
+from inspect import getfullargspec
+
+from models.user import User, Role
+
+
+class permission_required:
+    def __init__(self, permission):
+        self.permission = permission
+
+    def __call__(self, function):
+        @jwt_required()
+        @functools.wraps(function)
+        def wrapped_function(*args, **kwargs):
+            from app import DB_CONTROLLER
```

Comment:
is there a reason it can't be imported from outside once?

Reply:
Yes,, it's a circular dependency